### PR TITLE
Add missing methods to IonicaBizau/edit-json-file

### PR DESCRIPTION
**Pull request by AmannMax**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: 
- append() ->https://github.com/IonicaBizau/edit-json-file/blob/master/lib/index.js#L107
- pop() -> https://github.com/IonicaBizau/edit-json-file/blob/master/lib/index.js#L126
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/edit-json-file/index.d.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Type definitions for edit-json-file 1.4
+// Type definitions for edit-json-file 1.4.1
```
Comment (Twixes) [inline types/edit-json-file/index.d.ts:1]: Thank you for the update (though I haven't had anything to do with this library for a long time 😅), just this one thing is incorrect – looking at `edit-json-file` I see that the functions added here are 1.6.0+ features.
```suggestion
// Type definitions for edit-json-file 1.6.0
```

File: types/edit-json-file/edit-json-file-tests.ts [hunk 2]
```diff
@@ -3,6 +3,8 @@ import * as editJsonFile from 'edit-json-file';
 const jsonEditor: editJsonFile.JsonEditor = editJsonFile('1.json'); // $ExpectType JsonEditor
 jsonEditor.set('a', 2); // $ExpectType JsonEditor
 jsonEditor.set('b', []); // $ExpectType JsonEditor
+jsonEditor.append('a', 'b'); // $ExpectType JsonEditor
+jsonEditor.pop('a'); // $ExpectType any
```
Comment (Twixes) [inline types/edit-json-file/edit-json-file-tests.ts]: This returns `JsonEditor` in `index.d.ts` – which I confirmed in code of the library itself too – and is responsible for the CI failure.
```suggestion
jsonEditor.pop('a'); // $ExpectType JsonEditor
```

**Review by Twixes — CHANGES_REQUESTED** [review]

File: types/edit-json-file/index.d.ts (hunk 1 above)
Comment (AmannMax) [reply types/edit-json-file/index.d.ts:1]: Thank you!
I thougth i've to set it to the first version with this methods implemented.

**Review by Twixes — APPROVED** [review]
Nice

**Comment by AmannMax** [comment]
Ready to merge

**Pull request merged**